frontend/app/services/TouchpointBackend.scala

```diff
@@ -70,7 +70,15 @@ object TouchpointBackend {
 
     val catalogRestClient: SimpleClient[Future] = new SimpleClient[Future](restBackendConfig, RequestRunners.configurableFutureRunner(60.seconds))
     val newCatalogService = new subsv2.services.CatalogService[Future](pids, FetchCatalog.fromZuoraApi(catalogRestClient), Await.result(_, 60.seconds), restBackendConfig.envName)
-    val futureCatalog: Future[CatalogMap] = newCatalogService.catalog.map(_.fold[CatalogMap](error => {println(s"error: ${error.list.toList.mkString}"); Map()}, _.map))
+
+    val futureCatalog: Future[CatalogMap] = newCatalogService.catalog
+      .map(_.fold[CatalogMap](error => {println(s"error: ${error.list.toList.mkString}"); Map()}, _.map))
+      .recover {
+        case error =>
+          SafeLogger.error(scrub"Failed to load catalog from Zuora due to: $error")
+          throw error
```

Comment:
this change is good, I just don't understand what the existing println was doing in there

Reply:
looks like an "if not populated yet, println it and return empty" kind of hack.  Not sure if it was supposed to be shipped in that state though.